streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -532,6 +532,21 @@ public void onAssignment(Assignment assignment) {
         }
         this.partitionToTaskIds = partitionToTaskIds;
         this.partitionsByHostState = info.partitionsByHostState;
+        // only need to build when not coordinator
```

Comment:
This comment is a bit misunderstanding since `coordinator` usually refers to the server side coordinator although the the client's internal communication channel is also named `Coordinator`. How about

```
non-leaders of the group does not have the cluster metadata since only leader received it during the assignment phase, therefore they need to build it upon getting the assignment from the encoded assignment information.
```